File: templates/templates/EpoxyHello.Avalonia11/EpoxyHello.UI/ViewModels/MainViewModel.cs
using Epoxy;

namespace EpoxyHello.ViewModels;

[ViewModel]
public sealed class MainViewModel
{
    public Command Ready { get; }
    public string Title { get; private set; } = "";

    public MainViewModel()
    {
        // A handler for window loaded
        this.Ready = Command.Factory.Create(() =>
        {
            this.Title = "Hello Epoxy!";
            return default;
        });
    }
}

File: templates/templates/FSharp.EpoxyHello.Avalonia11/EpoxyHello.UI/ViewModels/MainViewModel.fs
namespace EpoxyHello.ViewModels

open Epoxy
open System

[<Sealed; ViewModel>]
type MainViewModel() as self =
    do
        // A handler for window loaded
        self.Ready <- Command.Factory.create(fun () -> async {
            self.Title <- "Hello Epoxy!"
        })

    member val Ready: Command = null with get, set
    member val Title: string = "" with get, set
